File: frontend/src/app/_models/client/client-response.dto.ts
import SimpleStatusEnum from "../enum/simple-status.enum";

export default class ClientResponseDto {
    id!: string;
    name!: string;
    status!: SimpleStatusEnum;
    createdAt!: Date;
    updatedAt!: Date;

    constructor(entity?: ClientResponseDto) {
        if (entity) {
            Object.assign(this, entity);
        }
    }
}

File: backend/cadastros/src/main/java/com/ecsolutions/cadastros/model/dtos/client/ClientResponseDto.java
package com.ecsolutions.cadastros.model.dtos.client;

import com.ecsolutions.cadastros.model.enums.SimpleStatusEnum;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ClientResponseDto {
    private UUID id;
    private String name;
    private SimpleStatusEnum status;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;

}
